lib/client.js

```diff
@@ -212,6 +212,7 @@ Client.prototype._setSequenceArgs = function(argsScheme, args) {
     return args;
   }
   for (var partIndex in argsScheme) {
+    partIndex = partIndex.replace('[]','')
```

Comment:
If it's such a specific scenario I guess we should guard it behind some option? I know it won't be an elegant solution, but it would fit both needs if the option will trigger the newly introduced behaviour and proceed "as is" if not provided...?